test/nlp/test_default.py

```diff
@@ -0,0 +1,81 @@
+from utils import get_toy_data_seqclassification, get_automl_settings
+
+
+def test_starting_point_not_in_search_space():
+    from flaml import AutoML
+
+    """
+        test starting_points located outside of the search space, and custom_hp is not set
+    """
+    X_train, y_train, X_val, y_val, _ = get_toy_data_seqclassification()
+
+    automl = AutoML()
+    automl_settings = get_automl_settings(estimator_name="transformer")
+
+    automl_settings["starting_points"] = {"transformer": [{"learning_rate": 2e-3}]}
+
+    automl.fit(X_train, y_train, **automl_settings)
+
+    """
+        test starting_points located outside of the search space, and custom_hp is set
+    """
+
+    from flaml import tune
+
+    X_train, y_train, X_val, y_val, _ = get_toy_data_seqclassification()
+
+    automl = AutoML()
+    automl_settings = get_automl_settings(estimator_name="transformer_ms")
+
+    automl_settings["custom_hp"] = {
+        "transformer_ms": {
+            "model_path": {
+                "domain": tune.choice(["albert-base-v2"]),
+            }
+        }
+    }
+    automl_settings["starting_points"] = "data:test/nlp/default/"
+    del automl_settings["custom_fit_kwargs"]["transformer_ms"]["model_path"]
```

Comment:
what if you don't delete it? Should you test that case as well?

Reply:
If I don't delete it, it'll go to the assertion error: https://github.com/microsoft/FLAML/blob/67ec7550cd2168a33d06c35a17acc91fd8a5ba3c/flaml/model.py#L444

I can add a test with try catch here.